### ISSUE TITLE
DataTable the first used filter will not be displayed when changed

### ISSUE THREAD

**Issue by DarkCTO** [plotly/dash#2492]
Thank you so much for helping improve the quality of Dash!

We do our best to catch bugs during the release process, but we rely on your help to find the ones that slip through.

**Describe your context**
No need, can be recreated right from Dash tutorial here [DataTable Filtering](https://dash.plotly.com/datatable/filtering)

**Describe the bug**
**Only on a newly loaded site please!**
in Advanced filter usage section, let's say you enter "> 10000000" without quotes (10 million) in **pop** column
The page is down by half, good, now copy the string "{pop} s> 10000000" without quotes
Switch to Write to filter_query option, and paste it in
Now if you increase or decrease the zeroes, the filter section on the table gets updated EXCEPT at 10 million!
The filtering still occurs (pages change), but it's not displayed in the data table filter zone

**Expected behavior**

a change in data table update filter_query, but a write in filter_query at that initial value does not update the table.

**Screenshots**

[See for yourself](https://imgur.com/a/gtBkpUn)
If you go to 100 million and go back to 10 million, filter still shows 100 million
If you go to 1 million and go back to 10 million, filter still shows 1 million

**Basically, if you change the filter in datatable first, then change it from other controls, the filter in datatable will not update at the exact query you used in the datatable.**

**Comment by AnnMarieW** [comment]
Since this was reported, DataTable has been deprecated, and we recommend using Dash AG Grid.

Dash AG Grid uses AG Grid's filter model to control filters, which can be read and updated from Dash callbacks. This avoids the synchronization issue described here between the DataTable filter UI and `filter_query`.

For more information see the Dash AG Grid docs:
 - [Filtering Overview]( https://dash.plotly.com/dash-ag-grid/filtering-overview)
- [Filter Model Callbacks ](https://dash.plotly.com/dash-ag-grid/callback-filters)



If you have questions about migrating to Dash AG Grid, please ask on the [Plotly Community Forum](https://community.plotly.com/).
